src/aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -633,7 +633,10 @@ def _perform_import(self):
                 with SandboxFolder(filepath_sandbox) as retrieved_temporary_folder:
                     self.presubmit(folder)
                     self.node.set_remote_workdir(self.inputs.remote_folder.get_remote_path())
-                    retrieve_calculation(self.node, transport, retrieved_temporary_folder.abspath)
+                    retrieved = retrieve_calculation(self.node, transport, retrieved_temporary_folder.abspath)
+                    if retrieved is not None:
+                        self.out(self.node.link_label_retrieved, retrieved)
+                        self.update_outputs()
```

Comment:
> At the end of those tasks, the process will change to a new state and the outputs will be "flushed" to the database by `update_outputs`.

Is `_perform_import` where this happens?

Reply:
No, this is the method that is called when a `CalcJobNode` is run in "import" mode. That is when a `remote_folder` is passed as an input, in which case the actual code is not run, but a `CalcJobNode` is created taking the outputs from the already existing `remote_folder`. The comment you quoted to the lifetime of an actual `CalcJob` process that is being run, which will call `update_outputs` on state changes.

Reply:
Where is it then actually calling `update_outputs`?

Reply:
https://github.com/aiidateam/aiida-core/blob/35d7ca63b44f051a26d3f96d84e043919eb3f101/src/aiida/engine/processes/process.py#L417